pkg/archive/archive.go

```diff
@@ -148,12 +149,18 @@ func DecompressStream(archive io.Reader) (io.ReadCloser, error) {
 		readBufWrapper := p.NewReadCloserWrapper(buf, bz2Reader)
 		return readBufWrapper, nil
 	case Xz:
-		xzReader, err := xzDecompress(buf)
+		xzReader, chdone, err := xzDecompress(buf)
 		if err != nil {
 			return nil, err
 		}
 		readBufWrapper := p.NewReadCloserWrapper(buf, xzReader)
-		return readBufWrapper, nil
+		return ioutils.NewReadCloserWrapper(readBufWrapper, func() error {
+			go func() {
```

Comment:
I am trying to change it like this, but it ends up a deadlock,

```go
                return ioutils.NewReadCloserWrapper(readBufWrapper, func() error {                                                                            
                        <-chdone                                                                                                                              
                        return readBufWrapper.Close()                                                                                                         
                }), nil
```

race can be detected by `go build -race` but how to debug a deadlock?